plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/refresh/TransportRefreshAction.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugin.ingest.transport.refresh;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugin.ingest.PipelineStore;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Refreshes the in-memory representaion of all the pipelines on all ingest nodes.
+ */
+public class TransportRefreshAction extends TransportAction<RefreshRequest, RefreshResponse> implements TransportRequestHandler<RefreshRequest> {
+
+    private final PipelineStore pipelineStore;
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+
+    @Inject
+    public TransportRefreshAction(Settings settings, ThreadPool threadPool, ActionFilters actionFilters, IndexNameExpressionResolver resolver, PipelineStore pipelineStore, ClusterService clusterService, TransportService transportService) {
+        super(settings, RefreshAction.NAME, threadPool, actionFilters, resolver);
+        this.pipelineStore = pipelineStore;
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        transportService.registerRequestHandler(RefreshAction.NAME, RefreshRequest::new, ThreadPool.Names.SAME, this);
```

Comment:
interesting, so it is completely unrelated to the threadpool of the response handler (in this case `ThreadPool.Names.MANAGEMENT`).

thanks